File: web/js/message.js
/**
 * Created by Cfw on 2016/10/2.
 */
var Message = function(userNameParam,typeParam,contentParam,countParam){
    var userName = "";
    var type = "";
    var content = "";
    var count = "";

    if("undefined"!=typeof(userNameParam) && userNameParam != null){
        userName = userNameParam;
    }
    if("undefined"!=typeof(typeParam) && typeParam != null){
        type = typeParam;
    }
    if("undefined"!=typeof(contentParam) && contentParam != null){
        content = contentParam;
    }
    if("undefined"!=typeof(countParam) && countParam != null){
        count = countParam;
    }

    this.setUserName = function(userNameParam){
        userName = userNameParam;
    }
    this.getUserName = function(){
        return userName;
    }
    this.setType = function(typeParam){
        type = typeParam;
    }
    this.getType = function(){
        return type;
    }
    this.setContent = function(contentParam){
        content = contentParam;
    }
    this.getContent = function(){
        return content;
    }
    this.setCount = function(countParam){
        count = countParam;
    }
    this.getCount = function(){
        return count;
    }

    this.getJson = function(){
        return '{"type":"'+type+'","content":"'+content+'","userName":"'+userName+'"}';
    }
}

var parseJson = function(jsonMessageStr){
    var jsonMessage = eval("("+jsonMessageStr+")");
    var message = new Message(jsonMessage.userName,jsonMessage.type,jsonMessage.content);

    return message;

}

Message.prototype.parseJson = parseJson;

File: src/main/java/spring/Model/Message.java
package spring.Model;

import com.google.gson.Gson;

/**
 * Created by Cfw on 2016/10/2.
 */
public class Message {

    private static Gson gson = new Gson();

    private String userName;

    private int type;

    private String content;

    private int count;

    public Message(){}

    public Message(int type, String content){
        this.type = type;
        this.content = content;
    }
    public Message(String jsonMessage){
        Message m = Message.gson.fromJson(jsonMessage,Message.class);
        this.userName = m.getUserName();
        this.type = m.getType();
        this.content = m.getContent();
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public String toString(){
        return Message.gson.toJson(this);
    }
}
